### PR TITLE
Plugin: The rules

### PR DESCRIPTION
**Pull request by meetmangukiya**

### CODE REVIEW

**Comment by Udayan12167** [comment]
Can you package both files in a folder inside the plugins folder. Further plugins might have multiple files. It would be better to package that way.

File: plugins/the_rules.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from errbot import BotPlugin, re_botcmd
+
+
+class The_rules(BotPlugin):
+    """
+    List the bot rules
+    """
+
+    RULES = [
+        "A robot may not harm humanity, or, by inaction, allow humanity to "
```
Comment (Udayan12167) [inline plugins/the_rules.py]: Double quotes to single quotes. Quotes bear not throwing errors.

File: plugins/the_rules.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "A robot must protect its own existence as long as such protection "
+        "does not conflict with the First or Second Law."
+    ]
+
+    @re_botcmd(pattern=r"the\s+rules", flags=re.IGNORECASE)
+    def the_rules(self, msg, args):
+        """
+        Show the bot rules.
+        """
+        return "\n".join(str(i) + ". " + j for i, j in enumerate(self.RULES))
```
Comment (Udayan12167) [inline plugins/the_rules.py]: Is it possible to do it with markdown support? Like are the bullets automatically detected as markdown?

File: plugins/the_rules.py (hunk 1 above)
Comment (meetmangukiya) [inline plugins/the_rules.py]: Quotes bear is set to double quotes though, it'll complain :/

File: plugins/the_rules.py (hunk 2 above)
Comment (meetmangukiya) [inline plugins/the_rules.py]: It is markdown, they will be rendered as order list, not bullets though since this is ordered list

**Comment by meetmangukiya** [comment]
> Can you package both files in a folder inside the plugins folder. Further plugins might have multiple files. It would be better to package that way.

I don't think any plugin will have multiple files. I don't see any plugins extending over 300 lines, but I have no problem with this, I'll make a folder for all the plugins in that case. I'll move the plugins already in master to follow the same convention :+1:

File: plugins/the_rules.py (hunk 1 above)
Comment (meetmangukiya) [inline plugins/the_rules.py]: Not applying here, since applied in other PR, will merge that first and then this, so there are less conflicts :+1:

Comment (meetmangukiya) [inline plugins/the_rules.py]: Done :+1:

File: plugins/the_rules.py (hunk 2 above)
Comment (meetmangukiya) [inline plugins/the_rules.py]: > Is it possible to do it with markdown support? Like are the bullets automatically detected as markdown?
  
1. We return valid markdown
2. gitter does the rendering of our markdown

so yes, it is rendered as markdown :+1:

**Comment by Udayan12167** [comment]
rebase needed.

**Comment by Udayan12167** [comment]
ack https://github.com/coala/corobo/pull/52/commits/6dec2cfb8fc63cdbbfd8a88e1406ed04fb7d50ec

**Comment by meetmangukiya** [comment]
@rultor merge!

**Pull request merged**